common/scala/src/main/scala/org/apache/openwhisk/core/database/cosmosdb/RxObservableImplicits.scala

```diff
@@ -34,8 +34,12 @@ private[cosmosdb] trait RxObservableImplicits {
      * @return the head result of the [[Observable]].
      */
     def head(): Future[T] = {
+      def toHandler[T](f: (T) => Unit): Action1[T] = new Action1[T] {
```

Comment:
May be simplify it further with SAM. Also using `P` instead of `T` as Intellij warned of shadowing here

```scala
def toHandler[P](f: (P) => Unit): Action1[P] = (t: P) => f(t)
```

Reply:
Good point! You can tell my scala is a little on the rusty side 😂